### PR TITLE
migrate `setup_environment.sh` to CMake

### LINKED ISSUE #750: Update `setup_environment.sh` to use CMake
Now that Bitcoin Core has CMake, it would be good to update `setup_environment.sh` to build Core with it.

### PR DESCRIPTION
**Pull request by brunoerg**
Fixes #750

### CODE REVIEW

File: test/setup_environment.sh [hunk 1]
```diff
@@ -448,7 +448,6 @@ if [[ -n ${build_bitcoind} ]]; then
     popd
 
     # Do the build
-    ./autogen.sh
-    CONFIG_SITE=$PWD/depends/x86_64-pc-linux-gnu/share/config.site ./configure --with-incompatible-bdb --with-miniupnpc=no --without-gui --disable-zmq --disable-tests --disable-bench --with-libs=no --with-utils=no
-    make src/bitcoind
+    cmake -B build -DBerkeleyDB_INCLUDE_DIR:PATH="$PWD/depends/x86_64-pc-linux-gnu/include" -DBoost_INCLUDE_DIR:PATH="$PWD/depends/x86_64-pc-linux-gnu/include" -DLibevent_DIR="$PWD/depends/x86_64-pc-linux-gnu/libevent"  -DWARN_INCOMPATIBLE_BDB=ON -DWITH_MINIUPNPC=OFF -DBUILD_GUI=OFF -DWITH_ZMQ=OFF -DBUILD_TESTS=OFF -DBUILD_BENCH=NO
```
Comment (hebasto) [inline test/setup_environment.sh:451]:
```suggestion
    cmake -B build --toolchain depends/x86_64-pc-linux-gnu/toolchain.cmake -DBUILD_TESTS=OFF -DBUILD_BENCH=OFF
```

Comment (hebasto) [reply test/setup_environment.sh:451]: `-DBUILD_GUI=OFF` is redundant when depends are built with `NO_QT=1` and so on.

Comment (brunoerg) [reply test/setup_environment.sh:451]: Cool, thanks @hebasto.

Comment (brunoerg) [reply test/setup_environment.sh]: Done

**Comment by brunoerg** [comment]
Thanks, @hebasto. Addressed https://github.com/bitcoin-core/HWI/pull/752#discussion_r1743650582. The solution is simpler than I thought.

**Comment by hebasto** [comment]
Not related to the CMake migration, but I'd suggest additionally this diff:
```diff
--- a/test/setup_environment.sh
+++ b/test/setup_environment.sh
@@ -444,7 +444,7 @@ if [[ -n ${build_bitcoind} ]]; then
 
     # Build bitcoind. This is super slow, but it is cached so it runs fairly quickly.
     pushd depends
-    make NO_QT=1 NO_QR=1 NO_ZMQ=1 NO_UPNP=1 NO_NATPMP=1
+    make NO_QT=1 NO_QR=1 NO_ZMQ=1 NO_UPNP=1 NO_NATPMP=1 NO_USDT=1
     popd
 
     # Do the build
```
to skip needless build of the `systemtap` package in depends.

**Comment by hebasto** [comment]
One more amendment is needed:
```diff
--- a/.github/actions/build-bitcoind/action.yml
+++ b/.github/actions/build-bitcoind/action.yml
@@ -23,7 +23,7 @@ runs:
         ccache --zero-stats
         cd test; ./setup_environment.sh --bitcoind; cd ..
         ccache --show-stats --verbose
-        tar -czf bitcoind.tar.gz test/work/bitcoin/src/bitcoind
+        tar -czf bitcoind.tar.gz test/work/bitcoin/build/src/bitcoind
 
     - uses: actions/cache/save@v4
       if: github.event_name != 'pull_request' && steps.ccache-cache.outputs.cache-hit != 'true'
```

**Comment by brunoerg** [comment]
Addressed https://github.com/bitcoin-core/HWI/pull/752#issuecomment-2328840413 and https://github.com/bitcoin-core/HWI/pull/752#issuecomment-2328877870. Thanks, @hebasto. Just added you as co-author.

File: test/setup_environment.sh [hunk 2]
```diff
@@ -444,11 +444,10 @@ if [[ -n ${build_bitcoind} ]]; then
...
-    make NO_QT=1 NO_QR=1 NO_ZMQ=1 NO_UPNP=1 NO_NATPMP=1
+    make NO_QT=1 NO_QR=1 NO_ZMQ=1 NO_UPNP=1 NO_NATPMP=1 NO_USDT=1
     popd
 
     # Do the build
-    ./autogen.sh
-    CONFIG_SITE=$PWD/depends/x86_64-pc-linux-gnu/share/config.site ./configure --with-incompatible-bdb --with-miniupnpc=no --without-gui --disable-zmq --disable-tests --disable-bench --with-libs=no --with-utils=no
-    make src/bitcoind
+    cmake -B build --toolchain depends/x86_64-pc-linux-gnu/toolchain.cmake -DBUILD_TESTS=OFF -DBUILD_BENCH=OFF
+    cmake --build build --target bitcoind
```
Comment (hebasto) [inline test/setup_environment.sh:452]: Are we interested in engaging in parallelism:
```suggestion
    cmake --build build -j $(nproc) --target bitcoind
```
?

If so, the same applies to building depends.

**Review by hebasto — APPROVED** [review]
~ACK 37d520d22228512e5b08c43fb10771df92c6dee0.~

oops, another path issue (

**Comment by brunoerg** [comment]
Addressed https://github.com/bitcoin-core/HWI/pull/752#issuecomment-2329013906

File: test/setup_environment.sh (hunk 2 above)
Comment (brunoerg) [reply test/setup_environment.sh:452]: I didn't address it because we did not have it previously, but I don't see why not engage parallelism.

**Review by hebasto — APPROVED** [review]
ACK 937c1ae93dad1294e84d9bbb0268ce78670f2eb1.

**Comment by hebasto** [comment]
Not related to this PR changes, but it seems https://github.com/Coldcard/firmware/commit/65b652a6f4c74864cb8e1714e0870c1c6e46e454 has to be considered to fix the CI.

**Comment by brunoerg** [comment]
> UPD. I suggest to leave it for a follow up PR.

Just opened #754 as draft.

File: test/setup_environment.sh (hunk 2 above)
Comment (brunoerg) [reply test/setup_environment.sh:452]: Addressed it in #755. I don't see why not parallelize it.

**Review by Thompson1985 — APPROVED** [review]

**Pull request merged**